src/components/map/use-map-tariff-zones.ts

```diff
@@ -0,0 +1,124 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Language, useTranslation } from '@atb/translations';
+import mapboxgl from 'mapbox-gl';
+import { getReferenceDataName } from '@atb/utils/reference-data';
+import { centroid } from '@turf/turf';
+import { type TariffZone, getTariffZones } from '@atb/modules/firebase';
+import useSWR from 'swr';
+
+const TARIFF_ZONE_SOURCE_ID = 'tariff-zones';
+const ZONE_BOUNDARY_LAYER_ID = 'zone-boundary-layer';
+const ZONE_NAMES_LAYER_ID = 'zone-names-layer';
+
+export const useMapTariffZones = async (
+  mapRef: React.MutableRefObject<mapboxgl.Map | undefined>,
+) => {
+  const { language } = useTranslation();
+  const [isZonesVisible, setIsZonesVisible] = useState(false);
+  const [isStyleLoaded, setIsStyleLoaded] = useState(false);
+  const { data } = useSWR('tariffZones', getTariffZones, {
```

Comment:
Should we wrap the `useSWR` call in a helper function to make this function body tidier?